sig-release/members.md

```diff
@@ -0,0 +1,8 @@
+## SIG Release Membership
```

Comment:
same here, this feels like it may turn into more administrivia than it's worth, but let's find out